iac-common/src/main/java/org/sonar/iac/common/checks/TextUtils.java

```diff
@@ -68,4 +73,22 @@ public static boolean isValueTrue(@Nullable Tree tree) {
   public static boolean isValueFalse(@Nullable Tree tree) {
     return isValue(tree, "false").isTrue();
   }
+
+  public static List<String> getListValueElements(@Nullable Tree tree){
+    if(tree instanceof ScalarTree){
+      return List.of(TextUtils.getValue(tree).orElse(""));
+    } else if (tree instanceof SequenceTree) {
+      return getValuesOfSequenceTree((SequenceTree) tree);
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  private static List<String> getValuesOfSequenceTree(SequenceTree tree) {
+    return tree.elements().stream()
+      .map(TextUtils::getListValueElements)
+      .flatMap(List::stream)
+      .collect(Collectors.toList());
+  }
```

Comment:
I'm not happy with the location. The methods handling Yaml trees. This utils class is also used by Terraform and was intended to handle all trees which implement the TextTree interface, which does not make use of these trees. I would prefer having a new utils class in the YAML package like `YamlTreeUtils`

Reply:
I would say that we don't need this method for the check you've implemented. The `isSet` predicate always gets a ScalarTree in the context of the `reportIfAnyItem` you can directly handle as a string. Can you confirm?